### ISSUE TITLE
Reproducible builds failing with error with build flag -mod=vendor

### ISSUE THREAD

**Issue by MatthewVance** [restic/restic#2956]
Output of `restic version`
--------------------------
N/A but it is: `restic 0.9.6 compiled with go1.13.4 on linux/amd64`

How did you run restic exactly?
-------------------------------
```
cd /home/me/dev/repos/github.com/restic/builder
git pull
sudo docker pull debian:stable
sudo docker build --no-cache -t restic/builder .
curl -L https://github.com/restic/restic/archive/v0.10.0.tar.gz -O
tar -xzvf v0.10.0.tar.gz
mkdir output 
restic_version=0.10.0
sudo docker run --rm --volume "$PWD/restic-${restic_version}/restic-${restic_version}:/restic" --volume "$PWD/output:/output" restic/builder
```

What backend/server/service did you use to store the repository?
----------------------------------------------------------------
N/A but I use REST via Rclone

Expected behavior
-----------------
Reproducible builds created

Actual behavior
---------------
Receive error: `build flag -mod=vendor only valid when using modules`

Steps to reproduce the behavior
-------------------------------
Attempt to build using instructions at https://github.com/restic/builder.

Do you have any idea what may have caused this?
-----------------------------------------------
Not sure. I think it my have to do with removing vendored dependencies in Change #2482.

Do you have an idea how to solve the issue?
-------------------------------------------
No, I wish I did.

Did restic help you today? Did it make you happy in any way?
------------------------------------------------------------
Yes, I'm a huge fan and proponent of Restic. I really appreciate the program and the community!

**Comment by rawtaz** [comment]
From what I can tell, removing the `-mod=vendor` in https://github.com/restic/builder/blob/master/Dockerfile#L40 solves the problem. Am I missing something or is it that simple? I'm guessing this flag is a relic from older Go versions, before we updated to 1.15.2 recently.

Below is the log from when I built it, using the same commands as the ones you used, but having removed that flag in the Dockerfile:

~~~
$ docker run --rm --volume "$PWD/restic-${restic_version}:/restic" --volume "$PWD/output:/output" restic/builder
go: downloading github.com/spf13/pflag v1.0.5
go: downloading golang.org/x/sync v0.0.0-20200625203802-6e8e738ad208
building with 2 workers
go: downloading github.com/spf13/cobra v1.0.0
go: downloading gopkg.in/tomb.v2 v2.0.0-20161208151619-d5d1b5820637
go: downloading github.com/spf13/cobra v1.0.0
go: downloading github.com/restic/chunker v0.4.0
go: downloading bazil.org/fuse v0.0.0-20200407214033-5883e5a4b512
go: downloading github.com/minio/sha256-simd v0.1.1
go: downloading github.com/Azure/azure-sdk-for-go v46.1.0+incompatible
go: downloading golang.org/x/crypto v0.0.0-20200820211705-5c72a883971a
go: downloading github.com/restic/chunker v0.4.0
go: downloading github.com/juju/ratelimit v1.0.1
go: downloading github.com/pkg/errors v0.9.1
go: downloading github.com/ncw/swift v1.0.52
go: downloading github.com/hashicorp/golang-lru v0.5.4
go: downloading golang.org/x/crypto v0.0.0-20200820211705-5c72a883971a
go: downloading bazil.org/fuse v0.0.0-20200407214033-5883e5a4b512
go: downloading golang.org/x/net v0.0.0-20200904194848-62affa334b73
go: downloading github.com/Azure/azure-sdk-for-go v46.1.0+incompatible
go: downloading golang.org/x/net v0.0.0-20200904194848-62affa334b73
go: downloading github.com/dchest/siphash v1.2.1
go: downloading golang.org/x/oauth2 v0.0.0-20200902213428-5d25da1a8d43
go: downloading google.golang.org/api v0.32.0
go: downloading github.com/cespare/xxhash v1.1.0
go: downloading google.golang.org/api v0.32.0
go: downloading golang.org/x/text v0.3.3
go: downloading github.com/cenkalti/backoff v2.2.1+incompatible
go: downloading github.com/cenkalti/backoff v2.2.1+incompatible
go: downloading github.com/pkg/sftp v1.12.0
go: downloading github.com/cpuguy83/go-md2man/v2 v2.0.0
go: downloading golang.org/x/text v0.3.3
go: downloading github.com/minio/minio-go/v6 v6.0.57
go: downloading github.com/minio/minio-go/v6 v6.0.57
go: downloading github.com/elithrar/simple-scrypt v1.3.0
go: downloading gopkg.in/yaml.v2 v2.3.0
go: downloading gopkg.in/yaml.v2 v2.3.0
go: downloading cloud.google.com/go v0.66.0
go: downloading github.com/kr/fs v0.1.0
go: downloading github.com/kr/fs v0.1.0
go: downloading github.com/dchest/siphash v1.2.1
go: downloading github.com/kurin/blazer v0.5.3
go: downloading github.com/json-iterator/go v1.1.10
go: downloading golang.org/x/sys v0.0.0-20200918174421-af09f7315aff
go: downloading github.com/russross/blackfriday/v2 v2.0.1
go: downloading github.com/mitchellh/go-homedir v1.1.0
go: downloading github.com/modern-go/reflect2 v1.0.1
go: downloading github.com/shurcooL/sanitized_anchor_name v1.0.0
go: downloading golang.org/x/sys v0.0.0-20200918174421-af09f7315aff
go: downloading gopkg.in/ini.v1 v1.61.0
go: downloading github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: downloading github.com/minio/md5-simd v1.1.0
go: downloading github.com/pkg/xattr v0.4.1
go: downloading github.com/pkg/xattr v0.4.1
go: downloading github.com/minio/md5-simd v1.1.0
go: downloading github.com/klauspost/cpuid v1.3.1
go: downloading github.com/googleapis/gax-go/v2 v2.0.5
go: downloading go.opencensus.io v0.22.4
go: downloading google.golang.org/grpc v1.32.0
go: downloading google.golang.org/grpc v1.32.0
go: downloading go.opencensus.io v0.22.4
go: downloading github.com/googleapis/gax-go/v2 v2.0.5
go: downloading github.com/golang/groupcache v0.0.0-20200121045136-8c9f03a8e57e
go: downloading github.com/golang/groupcache v0.0.0-20200121045136-8c9f03a8e57e
go: downloading github.com/golang/protobuf v1.4.2
go: downloading google.golang.org/genproto v0.0.0-20200918140846-d0d605568037
go: downloading google.golang.org/genproto v0.0.0-20200918140846-d0d605568037
go: downloading github.com/golang/protobuf v1.4.2
go: downloading google.golang.org/protobuf v1.25.0
go: downloading google.golang.org/protobuf v1.25.0
go: downloading github.com/Azure/go-autorest v14.2.0+incompatible
go: downloading github.com/Azure/go-autorest v14.2.0+incompatible
go: downloading github.com/satori/go.uuid v1.2.0
go: downloading github.com/satori/go.uuid v1.2.0
go: downloading github.com/Azure/go-autorest/autorest v0.11.6
go: downloading github.com/Azure/go-autorest/autorest v0.11.6
go: downloading github.com/Azure/go-autorest/tracing v0.6.0
go: downloading github.com/Azure/go-autorest/tracing v0.6.0
go: downloading github.com/Azure/go-autorest/logger v0.2.0
go: downloading github.com/Azure/go-autorest/autorest/adal v0.9.4
go: downloading github.com/Azure/go-autorest/logger v0.2.0
go: downloading github.com/Azure/go-autorest/autorest/adal v0.9.4
go: downloading github.com/dgrijalva/jwt-go v3.2.0+incompatible
go: downloading github.com/dgrijalva/jwt-go v3.2.0+incompatible
go: downloading github.com/Azure/go-autorest/autorest/date v0.3.0
go: downloading github.com/Azure/go-autorest/autorest/date v0.3.0
built freebsd/386 in 260.303s
built freebsd/amd64 in 263.575s
built freebsd/arm in 212.704s
built linux/386 in 210.138s
built linux/amd64 in 178.320s
built linux/arm in 230.652s
built linux/arm64 in 220.775s
built linux/ppc64le in 225.549s
built openbsd/386 in 230.017s
go: downloading github.com/inconshreveable/mousetrap v1.0.0
built openbsd/amd64 in 247.993s
built windows/386 in 217.547s
built windows/amd64 in 222.884s
built darwin/amd64 in 159.976s
build finished in 1479.644s
~~~

**Comment by MatthewVance** [comment]
Thanks @rawtaz. I confirmed that worked. I thought I tried that before submitting this issue, but perhaps I accidentally removed the run flag when trying eliminating `-mod=vendor`. I also noticed a mistake in my `docker run` command above so maybe that was my issue earlier. Either way, I was able to reproduce the build this time. Thanks for the extra set of eyes and submitting the PR to address this in the builder repo.
